[Tan Zheng Jie, Dickson] iP

dt-td opened the pull request:
# CS2103T Individual Project - Duke
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

### Duke frees your mind of having to remember things you need to do. It's,
* Text-based
* Easy to learn
* ~~**Fast**~~ ***Super fast*** to use

### All you need to do is,
1. Download the .jar file from [here](https://github.com/dt-td/ip)
2. In the directory of your .jar file, run this line in terminal `java -jar ip.jar`
3. Let it manage your tasks for you :smile: 

### And it is FREE!

### Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

#### If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```javascript
public static void main(String[] args) {
        new Duke("\\duke.txt").run();
}
```

Commit:
Add Gradle support

dt-td committed:
Add Level-1

dt-td committed:
Add Level-2

dt-td committed:
Add Level-3

dt-td committed:
Add Level-4

dt-td committed:
Add A-TextUiTesting

dt-td committed:
Add Level-5

dt-td committed:
Add Level-6

dt-td committed:
Add Level-7

dt-td committed:
Add Level-8

dt-td committed:
A-MoreOOP

dt-td committed:
Add A-Packages

dt-td committed:
Add A-Packages

dt-td committed:
Add J-Unit

dt-td committed:
A-JavaDoc

dt-td committed:
A-CodingStandard

dt-td committed:
Add Level-9

dt-td committed:
Merge branch 'branch-A-CodingStandard'

dt-td committed:
Merge branch 'branch-Level-9'

dt-td committed:
Cleaned up code

dt-td committed:
Cleaned up code

dt-td committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

dt-td committed:
Add A-Gradle

dt-td committed:
Add A-CheckStyle

ngjiayuan commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,178 @@
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
```
I like how you split the imports so neatly!

ngjiayuan commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+
+/**
+ * Encapsulates a deadline task.
+ * Deadline tasks are tasks that needs to be done by a specific date/time.
```
I like your javadoc description, it is very clear.

alissayarmantho replied:
+1

ngjiayuan commented on src/main/java/duke/exceptions/IllegalFormatException.java:
```diff
@@ -0,0 +1,13 @@
+ */
+public class IllegalFormatException extends DukeException {
+    /**
+     * Constructor for IllegalFormatException.
```
Did you miss out the @param for the javadoc?

ngjiayuan commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,300 @@
+    private String doneMessage(Task task) {
+        return "Nice! I've marked this task as done:\n" + task;
+    }
+}
```
I like how descriptive your javadocs are! Maybe you have forgotten some punctuations?

ngjiayuan reviewed (commented):
LGTM! Just some nits to fix.

dt-td committed:
Add Level-10

alissayarmantho commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,314 @@
+        if (!command.matches(regex)) {
+            throw new IllegalFormatException(correctFormat);
+        }
+    }
```
Not sure why you are checking the command format at TaskList.java. Shouldn't this be handled by Parser for separation of concern?

alissayarmantho commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,35 @@
+     * @return Command to run in duke bot.
+     * @throws UnknownCommandException invalid user command
+     */
+    public static Command parse(String command) throws UnknownCommandException {
```
You should probably handle command validity and regex checks here.

alissayarmantho commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,314 @@
+        String keyword = command.substring(5).trim();
+
+        List<Task> filteredTaskList = tasks.stream()
+                                           .filter(str -> str.toString().contains(keyword))
```
I like this code

alissayarmantho commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,178 @@
+     * @throws EmptyListException if TaskList is empty
+     */
+    public void saveFile(TaskList taskList) throws IOException, EmptyListException {
+        String path = System.getProperty("user.dir") + filePath;
```
Nice to know you are using relative path

dt-td committed:
Cleaned up code

dt-td committed:
Cleaned up code

Emoji shows up differently in gui.

Emoji not needed.

Let's remove emoji.

dt-td committed:
Add A-Assertions

dt-td committed:
Merge pull request #1 from dt-td/branch-A-Assertions

Add A-Assertions

dt-td committed:
Add A-CodeQuality

dt-td committed:
Merge pull request #2 from dt-td/branch-A-CodeQuality

Add A-CodeQuality

dt-td committed:
Add C-Help

No help page to assist users in usage of Duke bot

Users need help page if they are new users to know how to use Duke bot.

Let's,
* update start message of Duke bot to inform users to type "help" if they need assistance on the usage of Duke bot.
* add help command to list all the available functions in Duke bot.

This informs the user on the commands they can use with Duke bot.

dt-td committed:
Merge branch 'master' of https://github.com/dt-td/ip

dt-td committed:
Add A-CI

dt-td committed:
Update TaskList.java

Cleaned up code

dt-td committed:
Cleaned up code

dt-td committed:
Revert "Update TaskList.java"

This reverts commit 59ea17af21ec0f67dd086b16c92da4ade3b8acb8.

dt-td committed:
Revert "Revert "Update TaskList.java""

This reverts commit a3bef33edf456577214de0655071286b86e2106b.

dt-td committed:
Merge remote-tracking branch 'origin/master'

dt-td committed:
Add A-BetterGui

dt-td committed:
Cleaned up code.

dt-td committed:
Modified Ui

dt-td committed:
Add User Guide

dt-td committed:
Update README.md

dt-td committed:
Update README.md

dt-td committed:
Update README.md

dt-td committed:
Update README.md

dt-td committed:
Set theme jekyll-theme-slate

dt-td committed:
Set theme jekyll-theme-midnight

dt-td committed:
Update filename

dt-td committed:
Merge remote-tracking branch 'origin/master'

dt-td committed:
Merge branch 'master' of https://github.com/dt-td/ip

dt-td committed:
Reupload Ui.png

dt-td committed:
Upload Ui.png

dt-td committed:
Delete Ui.PNG

dt-td committed:
Reupload Ui.png

dt-td committed:
Cleaned up code

dt-td committed:
Cleaned up header comments

dt-td committed:
Set theme jekyll-theme-architect

dt-td committed:
Set theme jekyll-theme-architect

dt-td committed:
Update README.md

dt-td committed:
Update build.gradle

dt-td committed:
Set theme jekyll-theme-slate

dt-td committed:
Finalised Ui.

dt-td committed:
Update README.md

dt-td committed:
Update README.md

dt-td committed:
Update pngs

dt-td committed:
Finalised Ui and tidied up comments
